pkg/kvstore/events.go

```diff
@@ -81,6 +84,20 @@ func newWatcher(prefix string, chanSize int) *Watcher {
 	return w
 }
 
+// emit attempts to notify the watcher of an event within the given context.
+// returning false if the context is done before the event is emitted.
+func (w *Watcher) emit(ctx context.Context, scope string, event KeyValueEvent) bool {
+	queueStart := spanstat.Start()
+	var ok bool
+	select {
+	case <-ctx.Done():
```

Comment:
I'd suggest adding an extra case for `<-w.stopWatch`, so that it covers both the case in which the stop signal is given via the context, and via `watcher.Stop`, for the sake of generality.